### PR TITLE
Fixed gfx9+ breakage from "Merge user data nodes"

### PR DESCRIPTION
**Pull request by trenouf**
That change broke access to the vertex buffer in a VS that was merged
into TCS or GS, and to the streamout table in a GS that was merged into
a VS or TES. Fixed.

Change-Id: I0353f6cabd897ecdcba3c5f7e43170d1c4c2dac1

### CODE REVIEW

**Comment by JacobHeAMD** [comment]
retest this please

File: patch/llpcPatchEntryPointMutate.cpp [hunk 1]
```diff
@@ -408,14 +408,27 @@ FunctionType* PatchEntryPointMutate::GenerateEntryPointType(
...
-                // Only the first shader stage needs a vertex buffer table.
-                if ((m_pContext->GetShaderStageMask() & (ShaderStageToMask(m_shaderStage) - 1)) == 0)
+                // Only the vertex shader needs a vertex buffer table.
+                if (m_shaderStage == ShaderStageVertex)
                 {
+                    reserveVbTable = true;
+                }
+                else if (m_pContext->GetGfxIpVersion().major >= 9)
+                {
+                    // On gfx9+, the shader stage that the vertex shader is merged in to needs a vertex buffer
```
Comment (amdrexu) [inline patch/llpcPatchEntryPointMutate.cpp]: gfx9+ -> GFX9+

File: patch/llpcPatchEntryPointMutate.cpp [hunk 2]
```diff
@@ -426,11 +439,24 @@ FunctionType* PatchEntryPointMutate::GenerateEntryPointType(
                      (ShaderStageToMask(ShaderStageFragment) - ShaderStageToMask(m_shaderStage))) ==
                     ShaderStageToMask(m_shaderStage))
                 {
+                    reserveStreamOutTable = true;
+                }
+                else if (m_pContext->GetGfxIpVersion().major >= 9)
+                {
+                    // On gfx9+, the shader stage that the last shader is merged in to needs a stream out
```
Comment (amdrexu) [inline patch/llpcPatchEntryPointMutate.cpp]: gfx9+ -> GFX9+

**Comment by amdvlk-admin** [comment]
### Test summary for commit ae3a5f70dd98be4d1488e2a9c20230c977620775
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev 9a1f99f049f3b417242dc7c09b86c52fd7d9ceaf</li>
<li>SPVGEN: dev 64013f150c2d41965a3d36fa159b55fba52b727a</li>
<li>XGL: dev 4d91a4045078cf6884ee05007eac9eec2b881dbb</li>
<li>LLVM: amd-vulkan-dev 4305f776f1207189667a383e22661ccd1e99cd25</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 900/900 (100.0%)</li>
<li>Failed: 0/900 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 366444/366444 (100.0%)</li>
<li>Failed: 0/366444 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 366444/366444 (100.0%)</li>
<li>Failed: 0/366444 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by JacobHeAMD** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit d40fed38679126751038f3a3c3169460f2fb53f4
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev 9a1f99f049f3b417242dc7c09b86c52fd7d9ceaf</li>
<li>SPVGEN: dev 64013f150c2d41965a3d36fa159b55fba52b727a</li>
<li>XGL: dev 4d91a4045078cf6884ee05007eac9eec2b881dbb</li>
<li>LLVM: amd-vulkan-dev 4305f776f1207189667a383e22661ccd1e99cd25</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 900/900 (100.0%)</li>
<li>Failed: 0/900 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 366444/366444 (100.0%)</li>
<li>Failed: 0/366444 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 366444/366444 (100.0%)</li>
<li>Failed: 0/366444 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**